Content.Server/Antag/Components/AntagSelectionComponent.cs

```diff
@@ -0,0 +1,195 @@
+{
+    /// <summary>
+    /// Has the primary selection of antagonists finished yet?
+    /// </summary>
+    [DataField]
+    public bool SelectionsComplete;
+
+    /// <summary>
+    /// The definitions for the antagonists
+    /// </summary>
+    [DataField]
+    public List<AntagSelectionDefinition> Definitions = new();
+
+    /// <summary>
+    /// The minds and original names of the players selected to be antagonists.
+    /// </summary>
+    [DataField]
+    public List<(EntityUid, string)> SelectedMinds = new();
+
+    /// <summary>
+    /// When the antag selection will occur.
+    /// </summary>
+    [DataField]
+    public AntagSelectionTime SelectionTime = AntagSelectionTime.PostPlayerSpawn;
+
+    /// <summary>
+    /// Cached sessions of players who are chosen. Used so we don't have to rebuild the pool multiple times in a tick.
+    /// Is not serialized.
+    /// </summary>
+    public HashSet<ICommonSession> SelectedSessions = new();
+}
+
+[DataDefinition]
+public partial struct AntagSelectionDefinition()
+{
+    /// <summary>
+    /// A list of antagonist roles that are used for selecting which players will be antagonists.
+    /// </summary>
+    [DataField]
+    public List<ProtoId<AntagPrototype>> PrefRoles = new();
+
+    /// <summary>
+    /// Fallback for <see cref="PrefRoles"/>. Useful if you need multiple role preferences for a team antagonist.
+    /// </summary>
+    [DataField]
+    public List<ProtoId<AntagPrototype>> FallbackRoles = new();
+
+    /// <summary>
+    /// Should we allow people who already have an antagonist role?
+    /// </summary>
+    [DataField]
+    public AntagAcceptability MultiAntagSetting = AntagAcceptability.None;
+
+    /// <summary>
+    /// The minimum number of this antag.
+    /// </summary>
+    [DataField]
+    public int Min = 1;
+
+    /// <summary>
+    /// The maximum number of this antag.
+    /// </summary>
+    [DataField]
+    public int Max = 1;
+
+    /// <summary>
+    /// A range used to randomly select <see cref="Min"/>
+    /// </summary>
+    [DataField]
+    public MinMax? MinRange;
+
+    /// <summary>
+    /// A range used to randomly select <see cref="Max"/>
+    /// </summary>
+    [DataField]
+    public MinMax? MaxRange;
+
+    /// <summary>
+    /// a player to antag ratio: used to determine the amount of antags that will be present.
+    /// </summary>
+    [DataField]
+    public int PlayerRatio = 10;
+
+    /// <summary>
+    /// Whether or not players should be picked to inhabit this antag or not.
+    /// </summary>
+    [DataField]
+    public bool PickPlayer = true;
+
+    /// <summary>
+    /// If true, players that latejoin into a round have a chance of being converted into antagonists.
+    /// </summary>
+    [DataField]
+    public bool LateJoinAdditional = false;
+
+    //todo: find out how to do this with minimal boilerplate: filler department, maybe?
+    //public HashSet<ProtoId<JobPrototype>> JobBlacklist = new()
```

Comment:
Normally what I do for other stuff is have an intermediate datatype that lets you pick either a set of jobs or a prototype of a set of jobs, then also add the prototype.

So then you define the prototype in 1 spot and all the antag defs that want something generic like "no sec / command" just refer to that prototype instead.